pepper-apis/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java

```diff
@@ -337,10 +361,31 @@ private static Handle openJdbiWithAuthRetry(DB db) {
         try (Handle h = openJdbiWithAuthRetry(getDB())) {
             h.useTransaction(callback);
         } catch (ConnectionException e) {
-            throw new DDPException("Unable to get a connection from data source", e);
+            throw new DDPException(COULD_NOT_GET_CONNECTION, e);
         }
     }
 
+    private HikariDataSource createDataSource(int maxConnections, String dbUrl, DB db) {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(dbUrl);
+        config.addDataSourceProperty("cachePrepStmts", "true");
+        config.addDataSourceProperty("prepStmtCacheSize", "250");
+        config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+        config.addDataSourceProperty("cacheServerConfiguration", "true");
+        config.setInitializationFailTimeout(0); // don't fail to  create the pool if the db is not available
+        config.setAutoCommit(true); // will be managed by jdbi, which expects autcommit to be enabled initially
+        config.setTransactionIsolation("TRANSACTION_READ_COMMITTED");
+        config.setMaximumPoolSize(maxConnections);
+        config.setConnectionTimeout(1000);
+        config.setMaxLifetime((28800 * 1000) - (3 * 60 * 1000));  // as per hikari docs, a few minutes less than mysql wait_timeout
+        config.setPoolName(db.name());
+
+        // todo arz leverage allowPoolSuspension and mxbeans to fully  automate password rotation
+
+        return new HikariDataSource(config);
+    }
+
+
```

Comment:
A lot of changes here were merged from Zim's PR to fix issues found when running tests.